File: wallendar-app/app/src/main/java/com/td/wallendar/service/DebtsService.java
package com.td.wallendar.service;


import com.td.wallendar.dtos.response.DebtResponse;

import java.util.List;

import io.reactivex.Single;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface DebtsService {
    @GET("/api/users/{userId}/getDebts")
    Single<Response<List<DebtResponse>>> getTotalDebtsByUserId(@Path("userId") final Long userId);
}

File: wallendar-app/app/src/main/java/com/td/wallendar/service/DebtsService.kt
package com.td.wallendar.service

import com.td.wallendar.dtos.response.DebtResponse
import com.td.wallendar.models.DebtDetail
import io.reactivex.Single
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface DebtsService {
    @GET("/api/users/{userId}/getDebts")
    fun getTotalDebtsByUserId(@Path("userId") userId: Long): Single<Response<MutableList<DebtResponse>>>

    @GET("/api/debts/{debtId}/details")
    fun getDebtDetail(@Path("debtId") debtId: Long): Single<Response<MutableList<DebtDetail>>>
}